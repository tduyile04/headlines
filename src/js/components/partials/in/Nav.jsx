import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import HeadlineAction from '../../../actions/HeadlineAction';
import SideBar from './SideBar.jsx';


/**
 * Navigation bar display on the sources dashbooard
 * @class Nav
 * @extends {React.Component}
 */
class Nav extends React.Component {

  constructor() {
    super();
    this.state = {
      search: '',
      showAll: false
    };
    this.searchSources = this.searchSources.bind(this);
    this.onChange = this.onChange.bind(this);
    this.onKeyDown = this.onKeyDown.bind(this);
    this.logout = this.logout.bind(this);
  }

  /**
   * Creates search actions from collection of the sources in the store
   * @param {any} query
   * @memberof Nav
   */
  searchSources(query) {
    HeadlineAction.searchSources(query);
  }

  /**
   * Creates an action that gets data from the stores
   * @memberof Nav
   */
  retrieveAllSources() {
    HeadlineAction.getSources();
  }

  /**
   * Allows user input to be changed dynamically in the input field
   * @param {any} event
   * @memberof Nav
   */
  onChange(event) {
    this.setState({
      search: event.target.value
    });
  }

  /**
   * automates source search when the enter key is pressed
   * down
   * @param {any} event
   * @memberof Nav
   */
  onKeyDown(event) {
    const query = this.state.search;
    if (event.key === 'Enter') {
      this.searchSources(query);
      this.setState({
        search: ''
      });
    }
  }

  /**
   * logs out of the app and removes user profile from the local storage
   * @memberof Nav
   */
  logout() {
    localStorage.removeItem('userProfile');
    this.props.history.replace('/');
  }

  render() {
    const { user } = this.props;
    return (
      <nav>
        <div className="nav-wrapper grey darken-3">
          <div className="brand-logo">
            <SideBar user = {user} />
            <span
              className="white-text branded-logo"
              onClick={this.retrieveAllSources}
            >
              headlines
            </span>
          </div>
          <ul
            id="nav-mobile"
            className="right hide-on-med-and-down orange-text text-accent-1"
          >
            <li>
              <div className="input-field col s6 grey darken-3">
                <label htmlFor="search">
                  <i className="material-icons">search</i>
                </label>
                <input
                  type="search"
                  id="search"
                  value={this.state.search}
                  onChange={this.onChange}
                  onKeyDown = {this.onKeyDown}
                />
              </div>
            </li>
            <li><a onClick={this.logout}>Log Out</a></li>
          </ul>
        </div>
      </nav>
    );
  }
}

Nav.propTypes = {
  history: PropTypes.object,
  user: PropTypes.object
};

export default withRouter(Nav);
