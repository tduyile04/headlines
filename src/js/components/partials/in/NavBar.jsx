import React from 'react';
import { withRouter } from 'react-router-dom';
import SideBar from '../in/SideBar.jsx';

/**
 * Navigation bar display on the other dashboard
 * @class NavBar
 * @extends {React.Component}
 */
class NavBar extends React.Component {

  constructor() {
    super();
    this.state = {
      search: '',
    };
    this.logout = this.logout.bind(this);
  }

  /**
   * logs out of the app and removes user profile from the local storage
   * @memberof NavBar
   */
  logout() {
    localStorage.removeItem('userProfile');
    this.props.location.pathname = '/';
    location.reload();
  }

  render() {
    const { user } = this.props;
    return (
      <nav>
        <div className="nav-wrapper grey darken-3">
          <div className="brand-logo">
            <SideBar user={user} />
            <span
              className="white-text branded-logo"
              onClick={() => this.props.history.push('/sources')}
            >
              headlines
            </span>
          </div>
          <ul
            id="nav-mobile"
            className="right hide-on-med-and-down orange-text text-accent-1"
          >
            <li><a onClick={this.logout}>Log Out</a></li>
          </ul>
        </div>
      </nav>
    );
  }
}

export default withRouter(NavBar);
