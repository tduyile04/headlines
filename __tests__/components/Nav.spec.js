import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Nav from '../../src/js/components/partials/in/Nav.jsx';
import NavBar from '../../src/js/components/partials/in/NavBar.jsx';

describe('The Nav component', () => {
  it('should render correctly every time', () => {
    const wrapper = shallow(<Nav />);
    expect(wrapper).toHaveLength(1);
  });
  it('should return the expected user props', () => {
    const user = {
      name: 'jack doe',
      email: 'jackdoe@any.com',
      image: '/img/src/jack.jpg',
      idToken: '676tgybdy78h8e7ehue'
    };
    const wrapper = shallow(<Nav user={user} />);
    expect(wrapper.instance().props.user).toBe(user);
  });
  it('should render the Nav component correctly every time', () => {
    const userObject = {
      name: 'jack doe',
      email: 'jackdoe@any.com',
      image: '/img/src/jack.jpg',
      idToken: '676tgybdy78h8e7ehue'
    };
    const user = JSON.stringify(userObject);
    const tree = renderer.create(
      <MemoryRouter><Nav user={user} /></MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render the NavBar component correctly every time', () => {
    const userObject = {
      name: 'jack doe',
      email: 'jackdoe@any.com',
      image: '/img/src/jack.jpg',
      idToken: '676tgybdy78h8e7ehue'
    };
    const user = JSON.stringify(userObject);
    const tree = renderer.create(
      <MemoryRouter><NavBar user={user} /></MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
