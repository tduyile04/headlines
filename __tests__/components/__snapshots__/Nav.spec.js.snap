// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render a snapshot of the Nav Component 1`] = `
<nav>
  <div
    className="nav-wrapper grey darken-3"
  >
    <div
      className="brand-logo"
    >
      <span
        className="grey darken-3 sidebtn"
      >
        <span>
          <span
            className="grey darken-3"
            data-activates="sidenav_0"
          >
            <i
              className="material-icons white-text sidebtn"
            >
              menu
            </i>
          </span>
          <ul
            className="side-nav"
            id="sidenav_0"
          >
            <li
              className=""
            >
              <div
                className="userView"
              >
                <div
                  className="background"
                >
                  <img
                    src="../../../../images/cloudsidebar.jpg"
                  />
                </div>
                <a
                  href="#!user"
                >
                  <img
                    className="circle"
                    src="/img/src/jack.jpg"
                  />
                </a>
                <a
                  href="#!name"
                >
                  <span
                    className="white-text name"
                  >
                    jack doe
                  </span>
                </a>
                <a
                  href="#!email"
                >
                  <span
                    className="white-text email"
                  >
                    jackdoe@any.com
                  </span>
                </a>
              </div>
            </li>
            <li
              className=""
              onClick={[Function]}
            >
              <a
                className=""
                href="#!"
              >
                View Favourites
              </a>
            </li>
            <li
              className="divider"
            >
              <a
                className=""
                href="#!"
              />
            </li>
          </ul>
        </span>
      </span>
      <span
        className="white-text branded-logo"
        onClick={[Function]}
      >
        headlines
      </span>
    </div>
    <ul
      className="right hide-on-med-and-down orange-text text-accent-1"
      id="nav-mobile"
    >
      <li>
        <div
          className="input-field col s6 grey darken-3"
        >
          <label
            htmlFor="search"
          >
            <i
              className="material-icons"
            >
              search
            </i>
          </label>
          <input
            id="search"
            onChange={[Function]}
            onKeyDown={[Function]}
            type="search"
            value=""
          />
        </div>
      </li>
      <li>
        <a
          onClick={[Function]}
        >
          Log Out
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`<Nav /> should render a snapshot of the NavBar Component 1`] = `
<nav>
  <div
    className="nav-wrapper grey darken-3"
  >
    <div
      className="brand-logo"
    >
      <span
        className="grey darken-3 sidebtn"
      >
        <span>
          <span
            className="grey darken-3"
            data-activates="sidenav_1"
          >
            <i
              className="material-icons white-text sidebtn"
            >
              menu
            </i>
          </span>
          <ul
            className="side-nav"
            id="sidenav_1"
          >
            <li
              className=""
            >
              <div
                className="userView"
              >
                <div
                  className="background"
                >
                  <img
                    src="../../../../images/cloudsidebar.jpg"
                  />
                </div>
                <a
                  href="#!user"
                >
                  <img
                    className="circle"
                    src="/img/src/jack.jpg"
                  />
                </a>
                <a
                  href="#!name"
                >
                  <span
                    className="white-text name"
                  >
                    jack doe
                  </span>
                </a>
                <a
                  href="#!email"
                >
                  <span
                    className="white-text email"
                  >
                    jackdoe@any.com
                  </span>
                </a>
              </div>
            </li>
            <li
              className=""
              onClick={[Function]}
            >
              <a
                className=""
                href="#!"
              >
                View Favourites
              </a>
            </li>
            <li
              className="divider"
            >
              <a
                className=""
                href="#!"
              />
            </li>
          </ul>
        </span>
      </span>
      <span
        className="white-text branded-logo"
        onClick={[Function]}
      >
        headlines
      </span>
    </div>
    <ul
      className="right hide-on-med-and-down orange-text text-accent-1"
      id="nav-mobile"
    >
      <li>
        <a
          onClick={[Function]}
        >
          Log Out
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
